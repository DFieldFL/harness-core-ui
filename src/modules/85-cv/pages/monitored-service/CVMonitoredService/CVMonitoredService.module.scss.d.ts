/* eslint-disable */
// this is an auto-generated file
declare const styles: {
  readonly healthScoreCardContainer: string
  readonly healthScoreCard: string
  readonly tagsText: string
  readonly graphContainer: string
  readonly filterCard: string
  readonly filterSelect: string
  readonly pageBody: string
  readonly dependencyGraph: string
  readonly graphSummaryCard: string
  readonly infrastructureIconPadding: string
}
export default styles
