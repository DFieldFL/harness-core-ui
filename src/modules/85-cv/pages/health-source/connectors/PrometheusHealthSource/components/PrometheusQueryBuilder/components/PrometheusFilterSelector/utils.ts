import type { MultiSelectOption } from '@wings-software/uicore'

export function updateLabelOfSelectedFilter(
  selectedValue: string,
  item: MultiSelectOption,
  items: MultiSelectOption[]
): MultiSelectOption {
  const currentItem = items.find(listItem => listItem.value === item.value)
  if (currentItem) {
    const splitString = (currentItem.label as string).split(':')[0]
    return { ...currentItem, label: `${splitString}:${selectedValue}` }
  }

  return item
}

export function updateMultiSelectOption(
  newOption: MultiSelectOption,
  currentList: MultiSelectOption[]
): MultiSelectOption[] {
  if (!currentList?.length) {
    return []
  }

  const newOptionIndex = currentList.findIndex(listItem => listItem.value === newOption.value)
  if (newOptionIndex > -1) {
    currentList[newOptionIndex] = newOption
  }

  return Array.from(currentList)
}
