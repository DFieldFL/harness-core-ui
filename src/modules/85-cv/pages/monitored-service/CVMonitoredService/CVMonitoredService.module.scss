.healthScoreCardContainer {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  padding-left: var(--spacing-large);
}

.healthScoreCard {
  height: 31px;
  min-width: 34px;
  border: none !important;
  font-size: var(--font-size-small);
  text-align: center;
  font-weight: bold;
}

.tagsText {
  text-transform: capitalize;
}

.graphContainer {
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}

.filterCard {
  margin-right: var(--spacing-small);
  margin-bottom: var(--spacing-small);
  padding: var(--spacing-small) var(--spacing-large) !important;
}

.filterSelect {
  margin-right: var(--spacing-large);
}

$headerHeight: 146px;

.pageContainer {
  height: calc(100% - #{$headerHeight});
  overflow: auto;
  position: relative;
}

.pageBody {
  --page-header-height: #{$headerHeight} !important;
}

.noDataContainer {
  height: 100% !important;
}

.infrastructureIconPadding {
  margin-left: -3px !important ;
}

.noServiceAvailableImage {
  width: 720px;
  height: 350px;
  margin-right: 40px;
}

.linkButton {
  --button-height: auto !important;
  padding-left: 0 !important;
  width: fit-content;
}
