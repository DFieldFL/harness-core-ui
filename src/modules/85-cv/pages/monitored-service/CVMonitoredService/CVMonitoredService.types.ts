import type { MutateMethod } from 'restful-react'
import type { Color, FontVariation } from '@wings-software/uicore'
import type {
  MonitoredServiceListItemDTO,
  RestResponseHealthMonitoringFlagResponse,
  SetHealthMonitoringFlagQueryParams,
  SetHealthMonitoringFlagPathParams,
  PageMonitoredServiceListItemDTO
} from 'services/cv'
import type { FilterCardItem } from '@cv/components/FilterCard/FilterCard.types'
import type { DependencyData } from '@cv/components/DependencyGraph/DependencyGraph.types'

export interface FilterEnvInterface {
  searchTerm?: string
  environmentIdentifier?: string
}

export interface ContextMenuActionsProps {
  onEdit?(): void
  onDelete?(): void
  titleText: string
  contentText: string | JSX.Element
  confirmButtonText?: string
  deleteLabel?: string
  editLabel?: string
}

export type setHealthMonitoringFlagType = MutateMethod<
  RestResponseHealthMonitoringFlagResponse,
  void,
  SetHealthMonitoringFlagQueryParams,
  SetHealthMonitoringFlagPathParams
>

interface MonitoredServiceViewsProps {
  monitoredServiceListData?: PageMonitoredServiceListItemDTO
  refetchMonitoredServiceList: () => void
  selectedFilter?: FilterCardItem
  setSelectedFilter: React.Dispatch<React.SetStateAction<FilterCardItem | undefined>>
  onEditService: (identifier?: string) => void
  onDeleteService: (identifier?: string) => Promise<void>
  healthMonitoringFlagLoading?: boolean
  setHealthMonitoringFlag: setHealthMonitoringFlagType
}

export interface MonitoredServiceListViewProps extends MonitoredServiceViewsProps {
  setPage: React.Dispatch<React.SetStateAction<number>>
}

export interface MonitoredServiceGraphViewProps extends MonitoredServiceViewsProps {
  monitoredServiceDependencyData: DependencyData | null
}

export interface GraphSummaryCardProps {
  monitoredService: MonitoredServiceListItemDTO
  setHealthMonitoringFlag: setHealthMonitoringFlagType
  onEditService: (identifier?: string) => void
  onDeleteService: (identifier?: string) => Promise<void>
  healthMonitoringFlagLoading?: boolean
  refetchMonitoredServiceList: () => void
}

export interface ServiceHealthScoreProps {
  monitoredService: MonitoredServiceListItemDTO
  labelVariation?: FontVariation
  color?: Color
}
