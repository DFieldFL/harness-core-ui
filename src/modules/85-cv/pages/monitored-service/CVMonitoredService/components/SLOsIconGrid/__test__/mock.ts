import type { SloHealthIndicatorDTO } from 'services/cv'

export const items: SloHealthIndicatorDTO[] = [
  {
    serviceLevelObjectiveIdentifier: 'manager_http_errors',
    errorBudgetRemainingPercentage: 51.30434782608695,
    errorBudgetRisk: 'NEED_ATTENTION'
  },
  {
    serviceLevelObjectiveIdentifier: 'cvng_Slow_calls',
    errorBudgetRemainingPercentage: -71.99074074074075,
    errorBudgetRisk: 'UNHEALTHY'
  },
  {
    serviceLevelObjectiveIdentifier: 'slo',
    errorBudgetRemainingPercentage: -3111.8811881188117,
    errorBudgetRisk: 'HEALTHY'
  },
  {
    serviceLevelObjectiveIdentifier: 'slo-4',
    errorBudgetRemainingPercentage: -3111.8811881188117,
    errorBudgetRisk: 'OBSERVE'
  },
  {
    serviceLevelObjectiveIdentifier: 'slo-5',
    errorBudgetRemainingPercentage: 34.23,
    errorBudgetRisk: 'HEALTHY'
  }
]
