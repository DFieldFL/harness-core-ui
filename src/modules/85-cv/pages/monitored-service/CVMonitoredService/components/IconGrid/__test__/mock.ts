import type { RiskData } from 'services/cv'
import { RiskValues } from '@cv/utils/CommonUtils'

export const items: RiskData[] = [
  {
    riskStatus: RiskValues.HEALTHY,
    healthScore: 99
  },
  {
    riskStatus: RiskValues.OBSERVE,
    healthScore: 74
  },
  {
    riskStatus: RiskValues.NEED_ATTENTION,
    healthScore: 49
  },
  {
    riskStatus: RiskValues.NO_DATA
  },
  {
    riskStatus: RiskValues.UNHEALTHY,
    healthScore: 24
  }
]
