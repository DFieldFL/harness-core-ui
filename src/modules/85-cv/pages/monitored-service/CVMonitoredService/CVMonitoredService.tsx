import React, { useState, useMemo } from 'react'
import { useHistory, useParams } from 'react-router-dom'
import {
  Page,
  Button,
  ButtonVariation,
  Layout,
  Select,
  GridListToggle,
  Views,
  SelectOption,
  useToaster
} from '@wings-software/uicore'
import { useStrings } from 'framework/strings'
import {
  useGetMonitoredServiceListEnvironments,
  useListMonitoredService,
  useSetHealthMonitoringFlag,
  useDeleteMonitoredService,
  useGetServiceDependencyGraph
} from 'services/cv'
import routes from '@common/RouteDefinitions'
import { useQueryParams } from '@common/hooks'
import { NGBreadcrumbs } from '@common/components/NGBreadcrumbs/NGBreadcrumbs'
import type { ProjectPathProps } from '@common/interfaces/RouteInterfaces'
import { getCVMonitoringServicesSearchParam, getErrorMessage, getEnvironmentOptions } from '@cv/utils/CommonUtils'
import type { FilterCardItem } from '@cv/components/FilterCard/FilterCard.types'
import { getDependencyData } from '@cv/components/DependencyGraph/DependencyGraph.utils'
import { getFilterAndEnvironmentValue } from './CVMonitoredService.utils'
import MonitoredServiceListView from './components/MonitoredServiceListView/MonitoredServiceListView'
import MonitoredServiceGraphView from './components/MonitoredServiceGraphView/MonitoredServiceGraphView'
import css from './CVMonitoredService.module.scss'

const MonitoredService: React.FC = () => {
  const { getString } = useStrings()
  const history = useHistory()
  const { view } = useQueryParams<{ view?: Views.GRID }>()
  const { showError, clear } = useToaster()
  const params = useParams<ProjectPathProps>()
  const pathParams = {
    accountId: params.accountId,
    projectIdentifier: params.projectIdentifier,
    orgIdentifier: params.orgIdentifier
  }

  const [page, setPage] = useState(0)
  const [selectedView, setSelectedView] = useState<Views>(view === Views.GRID ? Views.GRID : Views.LIST)
  const [environment, setEnvironment] = useState<SelectOption>()
  const [selectedFilter, setSelectedFilter] = useState<FilterCardItem>()

  const { data: environmentDataList, loading: loadingEnvironments } = useGetMonitoredServiceListEnvironments({
    queryParams: pathParams
  })

  const {
    data: monitoredServiceListData,
    loading: monitoredServiceListLoading,
    refetch: refetchMonitoredServiceList,
    error: monitoredServiceListError
  } = useListMonitoredService({
    queryParams: {
      offset: page,
      pageSize: 10,
      ...pathParams,
      ...getFilterAndEnvironmentValue(environment?.value as string, '')
    },
    debounce: 400
  })

  const {
    data: serviceDependencyGraphData,
    loading: serviceDependencyGraphLoading,
    refetch: refetchServiceDependencyGraphData,
    error: serviceDependencyGraphError
  } = useGetServiceDependencyGraph({
    queryParams: {
      ...pathParams,
      ...getFilterAndEnvironmentValue(environment?.value as string, '')
    }
  })

  const monitoredServiceDependencyData = useMemo(
    () => getDependencyData(serviceDependencyGraphData),
    [serviceDependencyGraphData]
  )

  const { mutate: setHealthMonitoringFlag, loading: healthMonitoringFlagLoading } = useSetHealthMonitoringFlag({
    identifier: ''
  })

  const { mutate: deleteMonitoredService, loading: deleteMonitoredServiceLoading } = useDeleteMonitoredService({
    queryParams: pathParams
  })

  const onDeleteService = async (identifier?: string): Promise<void> => {
    if (!identifier) {
      return
    }

    const { pageIndex = 0, pageItemCount } = monitoredServiceListData?.data || {}

    try {
      const deletePromise = deleteMonitoredService(identifier)
      let refetchServiceListDataPromise, refetchServiceGraphDataPromise

      await deletePromise.then(() => {
        refetchServiceListDataPromise = refetchMonitoredServiceList()
        refetchServiceGraphDataPromise = refetchServiceDependencyGraphData()
      })

      await Promise.all([deletePromise, refetchServiceListDataPromise, refetchServiceGraphDataPromise])

      if (pageIndex > 0 && pageItemCount === 1) {
        setPage(page - 1)
      }
    } catch (e) {
      clear()
      showError(getErrorMessage(e))
    }
  }

  const onEditService = (identifier?: string): void => {
    history.push({
      pathname: routes.toCVMonitoredServiceConfigurations({
        ...pathParams,
        identifier: identifier,
        module: 'cv'
      }),
      search: getCVMonitoringServicesSearchParam(selectedView)
    })
  }

  return (
    <>
      <Page.Header breadcrumbs={<NGBreadcrumbs />} title={getString('cv.monitoredServices.title')} />
      <Page.Header
        title={
          <Button
            variation={ButtonVariation.PRIMARY}
            icon="plus"
            text={getString('cv.monitoredServices.newMonitoredServices')}
            onClick={() => {
              history.push({
                pathname: routes.toCVAddMonitoringServicesSetup(pathParams),
                search: getCVMonitoringServicesSearchParam(selectedView)
              })
            }}
          />
        }
        toolbar={
          <Layout.Horizontal>
            <Select
              value={{
                label: `${getString('environment')}: ${environment?.label ?? getString('all')}`,
                value: environment?.value ?? getString('all')
              }}
              defaultSelectedItem={{ label: getString('all'), value: getString('all') }}
              items={getEnvironmentOptions(environmentDataList, loadingEnvironments, getString)}
              onChange={item => {
                setSelectedFilter(undefined)
                setEnvironment(item)
              }}
              className={css.filterSelect}
            />
            <GridListToggle
              initialSelectedView={selectedView}
              onViewToggle={setSelectedView}
              icons={{ left: 'graph' }}
            />
          </Layout.Horizontal>
        }
      />
      {selectedView === Views.LIST ? (
        <Page.Body
          loading={monitoredServiceListLoading || deleteMonitoredServiceLoading || healthMonitoringFlagLoading}
          error={getErrorMessage(monitoredServiceListError)}
          retryOnError={() => refetchMonitoredServiceList()}
          noData={{
            when: () => !monitoredServiceListData?.data?.content?.length,
            icon: 'join-table',
            message: getString('cv.monitoredServices.noData')
          }}
          className={css.pageBody}
        >
          <MonitoredServiceListView
            monitoredServiceListData={monitoredServiceListData?.data}
            refetchMonitoredServiceList={refetchMonitoredServiceList}
            selectedFilter={selectedFilter}
            setSelectedFilter={setSelectedFilter}
            onEditService={onEditService}
            onDeleteService={onDeleteService}
            setHealthMonitoringFlag={setHealthMonitoringFlag}
            setPage={setPage}
          />
        </Page.Body>
      ) : (
        <Page.Body
          loading={
            serviceDependencyGraphLoading ||
            monitoredServiceListLoading ||
            deleteMonitoredServiceLoading ||
            healthMonitoringFlagLoading
          }
          error={getErrorMessage(serviceDependencyGraphError ?? monitoredServiceListError)}
          retryOnError={() => {
            if (serviceDependencyGraphError) {
              refetchServiceDependencyGraphData()
            }
            if (monitoredServiceListError) {
              refetchMonitoredServiceList()
            }
          }}
          noData={{
            when: () => !monitoredServiceDependencyData,
            message: getString('cv.monitoredServices.noData')
          }}
          className={css.pageBody}
        >
          <MonitoredServiceGraphView
            monitoredServiceListData={monitoredServiceListData?.data}
            refetchMonitoredServiceList={refetchMonitoredServiceList}
            selectedFilter={selectedFilter}
            setSelectedFilter={setSelectedFilter}
            onEditService={onEditService}
            onDeleteService={onDeleteService}
            setHealthMonitoringFlag={setHealthMonitoringFlag}
            healthMonitoringFlagLoading={healthMonitoringFlagLoading}
            monitoredServiceDependencyData={monitoredServiceDependencyData}
          />
        </Page.Body>
      )}
    </>
  )
}

export default MonitoredService
