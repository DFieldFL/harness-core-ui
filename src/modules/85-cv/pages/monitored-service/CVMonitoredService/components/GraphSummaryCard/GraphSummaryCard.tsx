import React from 'react'
import { Container, Layout, Text, Color, FontVariation, ButtonVariation, Button } from '@wings-software/uicore'
import { useStrings } from 'framework/strings'
import { useConfirmationDialog } from '@common/exports'
import ToggleMonitoring from '../../../components/toggleMonitoring/ToggleMonitoring'
import { ServiceDeleteContext, ServiceHealthTrend, ServiceHealthScore } from '../../CVMonitoredService.utils'
import type { GraphSummaryCardProps } from '../../CVMonitoredService.types'
import { GraphServiceChanges } from './GraphSummaryCard.utils'
import css from '../../CVMonitoredService.module.scss'

const GraphSummaryCard: React.FC<GraphSummaryCardProps> = ({
  monitoredService,
  setHealthMonitoringFlag,
  onEditService,
  onDeleteService,
  healthMonitoringFlagLoading,
  refetchMonitoredServiceList
}) => {
  const { getString } = useStrings()

  const { openDialog: confirmServiceDelete } = useConfirmationDialog({
    titleText: getString('common.delete', { name: monitoredService.serviceName }),
    contentText: <ServiceDeleteContext serviceName={monitoredService.serviceName} />,
    confirmButtonText: getString('yes'),
    cancelButtonText: getString('cancel'),
    onCloseDialog: function (shouldDelete: boolean) {
      if (shouldDelete) {
        onDeleteService(monitoredService.identifier)
      }
    }
  })

  return (
    <Container width={358} background={Color.GREY_700} padding="large" className={css.graphSummaryCard}>
      <Layout.Horizontal flex={{ justifyContent: 'space-between', alignItems: 'flex-start' }}>
        <Container>
          <Text color={Color.GREY_0} font={{ variation: FontVariation.H6 }}>
            {monitoredService?.serviceName?.toUpperCase()}
          </Text>
          <Text color={Color.GREY_0} font={{ variation: FontVariation.SMALL }}>
            {monitoredService?.environmentName}
          </Text>
        </Container>
        <Layout.Horizontal flex={{ alignItems: 'center' }}>
          <Button
            icon="Edit"
            iconProps={{ color: Color.GREY_0 }}
            variation={ButtonVariation.ICON}
            onClick={() => onEditService(monitoredService.identifier)}
          />
          <Button
            icon="trash"
            iconProps={{ color: Color.GREY_0 }}
            variation={ButtonVariation.ICON}
            margin={{ right: 'small' }}
            onClick={confirmServiceDelete}
          />
          <ToggleMonitoring
            refetch={refetchMonitoredServiceList}
            identifier={monitoredService?.identifier as string}
            enabled={!!monitoredService?.healthMonitoringEnabled}
            setHealthMonitoringFlag={setHealthMonitoringFlag}
            loading={healthMonitoringFlagLoading}
          />
        </Layout.Horizontal>
      </Layout.Horizontal>

      <Container
        border={{ top: true, bottom: true, color: Color.GREY_400 }}
        padding={{ top: 'medium', bottom: 'medium' }}
        margin={{ top: 'medium', bottom: 'medium' }}
      >
        <Text color={Color.GREY_0} font={{ variation: FontVariation.TINY_SEMI }} padding={{ bottom: 'medium' }}>
          {getString('cv.Dependency.serviceChanges').toUpperCase()}
        </Text>
        <GraphServiceChanges changeSummary={monitoredService.changeSummary} />
      </Container>

      <Layout.Vertical spacing="large">
        <Text color={Color.GREY_0} font={{ variation: FontVariation.TINY_SEMI }}>
          {getString('cv.monitoredServices.monitoredServiceTabs.serviceHealth').toUpperCase()}
        </Text>
        <ServiceHealthTrend healthScores={monitoredService.historicalTrend?.healthScores} />
        <ServiceHealthScore
          monitoredService={monitoredService}
          labelVariation={FontVariation.SMALL}
          color={Color.GREY_0}
        />
      </Layout.Vertical>
    </Container>
  )
}

export default GraphSummaryCard
