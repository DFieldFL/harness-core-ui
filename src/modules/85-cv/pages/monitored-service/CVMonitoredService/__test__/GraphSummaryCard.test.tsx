import React from 'react'
import userEvent from '@testing-library/user-event'
import { render, queryByText, waitFor } from '@testing-library/react'
import { findDialogContainer, TestWrapper } from '@common/utils/testUtils'
import { testWrapperProps } from '@cv/pages/monitored-service/CVMonitoredService/__test__/CVMonitoredService.test'
import GraphSummaryCard from '../components/GraphSummaryCard/GraphSummaryCard'
import { monitoredService } from './CVMonitoredService.mock'

const onToggleService = jest.fn()
const onEditService = jest.fn()
const onDeleteService = jest.fn()

describe('GraphSummaryCard', () => {
  test('Render GraphSummaryCard', () => {
    const { container } = render(
      <TestWrapper {...testWrapperProps}>
        <GraphSummaryCard
          monitoredService={monitoredService}
          onEditService={onEditService}
          onDeleteService={onDeleteService}
          onToggleService={onToggleService}
        />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })

  test('onEditService', async () => {
    const { container } = render(
      <TestWrapper {...testWrapperProps}>
        <GraphSummaryCard
          monitoredService={monitoredService}
          onEditService={onEditService}
          onDeleteService={onDeleteService}
          onToggleService={onToggleService}
        />
      </TestWrapper>
    )

    userEvent.click(container.querySelectorAll('[data-icon="Edit"]')[0])

    expect(onEditService).toBeCalledWith(monitoredService.identifier)
  })

  test('onDeleteService', async () => {
    const { container } = render(
      <TestWrapper {...testWrapperProps}>
        <GraphSummaryCard
          monitoredService={monitoredService}
          onEditService={onEditService}
          onDeleteService={onDeleteService}
          onToggleService={onToggleService}
        />
      </TestWrapper>
    )

    userEvent.click(container.querySelector('[data-icon="trash"]')!)

    await waitFor(() => queryByText(document.body, `yes`))
    const confirmDialog = findDialogContainer()
    expect(confirmDialog).toBeInTheDocument()

    const confirmButton = queryByText(confirmDialog!, 'yes')
    userEvent.click(confirmButton!)

    expect(onDeleteService).toBeCalledWith(monitoredService.identifier)
  })
})
