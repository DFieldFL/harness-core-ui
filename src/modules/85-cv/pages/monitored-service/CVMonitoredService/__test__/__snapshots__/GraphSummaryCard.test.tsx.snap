// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphSummaryCard Render GraphSummaryCard 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main graphSummaryCard StyledProps--background-grey700 StyledProps--padding-large"
    style="width: 358px;"
  >
    <div
      class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-space-between StyledProps--flex-alignItems-flex-start"
    >
      <div
        class="StyledProps--font StyledProps--main"
      >
        <p
          class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey0 StyledProps--font-variation-h6"
        >
          SERVICENAME
        </p>
        <p
          class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey0 StyledProps--font-variation-small"
        >
          environmentName
        </p>
      </div>
      <div
        class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-alignItems-center"
      >
        <button
          class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--withLeftIcon Button--variation Button--variation-icon"
          type="button"
        >
          <span
            class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey0"
            data-icon="Edit"
          >
            <svg
              height="16"
              viewBox="0 0 12 12"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fill-rule="evenodd"
                stroke-linecap="round"
                stroke-linejoin="round"
              >
                <path
                  d="M8.66 1.823h0L2.244 8.239v1.75h1.75l6.416-6.416a1.75 1.75 0 00-1.75-1.75z"
                  stroke="#555770"
                />
                <path
                  d="M6.583 5.146l2.063-2.063"
                  opacity="0.3"
                  stroke="#606170"
                />
              </g>
            </svg>
          </span>
        </button>
        <button
          class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--margin-right-small Button--with-current-color Button--iconOnly Button--withLeftIcon Button--variation Button--variation-icon"
          type="button"
        >
          <span
            class="bp3-icon bp3-icon-trash StyledProps--main StyledProps--color StyledProps--color-grey0"
            icon="trash"
          >
            <svg
              data-icon="trash"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                trash
              </desc>
              <path
                d="M14.49 3.99h-13c-.28 0-.5.22-.5.5s.22.5.5.5h.5v10c0 .55.45 1 1 1h10c.55 0 1-.45 1-1v-10h.5c.28 0 .5-.22.5-.5s-.22-.5-.5-.5zm-8.5 9c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm3 0c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm3 0c0 .55-.45 1-1 1s-1-.45-1-1v-6c0-.55.45-1 1-1s1 .45 1 1v6zm2-12h-4c0-.55-.45-1-1-1h-2c-.55 0-1 .45-1 1h-4c-.55 0-1 .45-1 1v1h14v-1c0-.55-.45-1-1-1z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </button>
        <div
          class="optionBtn"
        >
          <div
            class="item active disabled"
            data-name="on-btn"
            role="button"
          >
            common.ON
          </div>
          <div
            class="item"
            data-name="off-btn"
            role="button"
          >
            common.OFF
          </div>
        </div>
      </div>
    </div>
    <div
      class="StyledProps--font StyledProps--main StyledProps--border StyledProps--border-top StyledProps--border-bottom StyledProps--border-color-grey400 StyledProps--padding-top-medium StyledProps--padding-bottom-medium StyledProps--margin-top-medium StyledProps--margin-bottom-medium"
    >
      <p
        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey0 StyledProps--font-variation-tiny-semi StyledProps--padding-bottom-medium"
      >
        CV.DEPENDENCY.SERVICECHANGES
      </p>
      <div
        class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
      >
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-variation-small StyledProps--color StyledProps--color-grey100"
          style="display: flex; align-items: center;"
        >
          <span
            class="bp3-icon StyledProps--main Text--icon StyledProps--padding-right-xsmall StyledProps--color StyledProps--color-grey100"
            data-icon="nav-project"
          >
            <svg
              height="12"
              viewBox="0 0 22 22"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1.1 4.583L11 0l9.9 4.583v12.834L11 22l-9.9-4.583zm9 5.171l-7.2-3.33v9.811l7.2 3.338V9.754zm9 6.481V6.424l-7.2 3.33v9.819zm-1.462-11.15L11 2.012 4.362 5.085 11 8.157z"
                fill="#56778d"
              />
            </svg>
          </span>
          10 Deployments
        </p>
        <p
          class="StyledProps--font StyledProps--main infrastructureIconPadding StyledProps--font-variation-small StyledProps--color StyledProps--color-grey100"
          style="display: flex; align-items: center;"
        >
          <span
            class="bp3-icon StyledProps--main Text--icon StyledProps--padding-none StyledProps--color StyledProps--color-grey100"
            data-icon="infrastructure"
          >
            <svg
              fill="currentColor"
              height="18"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 4h16v16H4z"
                opacity="0.01"
              />
              <path
                clip-rule="evenodd"
                d="M12.487 8.343a.2.2 0 00.08.17l6.287 4.692a1 1 0 01.006 1.598l-6.257 4.74a1 1 0 01-1.207.002l-6.325-4.773a1 1 0 01-.026-1.577l.664-.536a.2.2 0 01.248-.002l.207.161a.3.3 0 01-.006.479l-.593.436a.2.2 0 00-.002.32l5.618 4.253a.2.2 0 00.321-.159v-2.515L5.066 10.8a1 1 0 010-1.6l6.344-4.758a1 1 0 011.187-.01l6.408 4.646a1 1 0 01-.036 1.645l-.905.597-.482-.471a.2.2 0 01.037-.315l.747-.449a.2.2 0 00.011-.335l-5.471-3.825a.2.2 0 00-.315.155l-.104 2.263zm-.767-2.476a.2.2 0 00-.32-.16L5.972 9.775a.2.2 0 000 .32l5.753 4.346a.5.5 0 00.567.025l2.642-1.65a.2.2 0 01.233.014l.452.37a.2.2 0 01-.02.324l-3.217 2.02a.2.2 0 00-.093.17v2.468a.2.2 0 00.322.158l5.42-4.18a.2.2 0 00-.001-.318l-5.853-4.469a.2.2 0 00-.245.002l-2.438 1.92a.2.2 0 01-.258-.008l-.382-.346a.2.2 0 01.011-.305l2.78-2.186a.2.2 0 00.075-.158V5.867z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          10 infrastructureText changes
        </p>
        <p
          class="StyledProps--font StyledProps--main StyledProps--font-variation-small StyledProps--color StyledProps--color-grey100"
          style="display: flex; align-items: center;"
        >
          <span
            class="bp3-icon StyledProps--main Text--icon StyledProps--padding-right-xsmall StyledProps--color StyledProps--color-grey100"
            data-icon="warning-outline"
          >
            <svg
              fill="#fff"
              height="12"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10.29 3.86L1.82 18a2 2 0 001.71 3h16.94a2 2 0 001.71-3L13.71 3.86a2 2 0 00-3.42 0zM12 9v4M12 17h.01"
              />
            </svg>
          </span>
          10 cv.changeSource.tooltip.incidents
        </p>
      </div>
    </div>
    <div
      class="Layout--vertical Layout--layout-spacing-large StyledProps--main"
    >
      <p
        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey0 StyledProps--font-variation-tiny-semi"
      >
        CV.MONITOREDSERVICES.MONITOREDSERVICETABS.SERVICEHEALTH
      </p>
      <div
        class="Layout--horizontal Layout--layout-spacing-small StyledProps--main healthScoreCardContainer"
      >
        <span
          class="Tag--main"
        >
          <span
            class="bp3-tag healthScoreCard"
          >
            <span
              class="bp3-text-overflow-ellipsis bp3-fill"
            >
              45
            </span>
          </span>
        </span>
        <p
          class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey0 StyledProps--font-variation-small"
        >
          cv.monitoredServices.serviceHealth.serviceDependencies.states.needsAttention
        </p>
      </div>
    </div>
  </div>
</div>
`;
